converters/fromZigbee.js

```diff
@@ -328,6 +328,122 @@ const tuyaThermostat = (model, msg, publish, options, meta) => {
     }
 };
 
+const tuyaThermostat2 = (model, msg, publish, options, meta) => {
```

Comment:
I don't understand why you created a new function here,  the only diff I see is you added:
```
case 1029: // fan mode 0 - low , 1 - medium , 2 - high , 3 - auto ( tested on 6dfgetq TUYA zigbee module )
        return {fan_mode: common.TuyaFanModes[dataAsDecNumber]};
```

which should also fit in `tuyaThermostat()`

Reply:
I do think the tuya specific function should be refacted to someway which can deal with different meanings across venders, since return 1 may mean open while may mean close in another vendor.
Btw , I am not familiar with js so i can help only a little here.
Hope this project will become a giant oneday!